unified-runtime/source/loader/layers/sanitizer/msan/msan_interceptor.cpp

```diff
@@ -638,47 +644,30 @@ ContextInfo::~ContextInfo() {
 ur_result_t USMLaunchInfo::initialize() {
   UR_CALL(getContext()->urDdiTable.Context.pfnRetain(Context));
   UR_CALL(getContext()->urDdiTable.Device.pfnRetain(Device));
-  UR_CALL(getContext()->urDdiTable.USM.pfnSharedAlloc(
-      Context, Device, nullptr, nullptr, sizeof(MsanLaunchInfo),
-      (void **)&Data));
-  *Data = MsanLaunchInfo{};
   return UR_RESULT_SUCCESS;
 }
 
-USMLaunchInfo::~USMLaunchInfo() {
-  [[maybe_unused]] ur_result_t Result;
-  if (Data) {
-    if (Data->CleanShadow) {
-      Result = getContext()->urDdiTable.USM.pfnFree(Context,
-                                                    (void *)Data->CleanShadow);
-      assert(Result == UR_RESULT_SUCCESS);
-    }
-    Result = getContext()->urDdiTable.USM.pfnFree(Context, (void *)Data);
+MsanRuntimeDataWrapper::~MsanRuntimeDataWrapper() {
+  if (Host.CleanShadow) {
+    auto Result =
```

Comment:
I think we still need the `[[maybe_unused]]` for this and the declaration below, assert doesn't always count as using a variable